go/arrow/csv/writer.go

```diff
@@ -44,6 +46,12 @@ func NewWriter(w io.Writer, schema *arrow.Schema, opts ...Option) *Writer {
 		opt(ww)
 	}
 
+	if ww.withHeader {
+		if err := ww.writeHeader(); err != nil {
```

Comment:
here as well, I'd opt for moving this to the `Write()` method, guarded with a `sync.Once`.